File: backend/app/routers/fishing.py
"""fishing 라우터 — 청계천 낚시터 (캐스팅/낚기/도감/판매)."""
from fastapi import APIRouter, Depends, HTTPException

from app.auth import get_current_user
from app.models.schemas import FishingLand, FishingSell
from app.services import fishing

router = APIRouter()


@router.get("/pond")
def pond(user: dict = Depends(get_current_user)):
    return fishing.pond_status()


@router.post("/cast")
def cast(user: dict = Depends(get_current_user)):
    return fishing.cast(user["id"])


@router.post("/land")
def land(body: FishingLand, user: dict = Depends(get_current_user)):
    try:
        return fishing.land(user["id"], body.token)
    except ValueError as e:
        raise HTTPException(status_code=400, detail=str(e))


@router.get("/collection")
def collection(user: dict = Depends(get_current_user)):
    return fishing.collection(user["id"])


@router.post("/sell")
def sell(body: FishingSell, user: dict = Depends(get_current_user)):
    try:
        return fishing.sell(user["id"], body.fish_id, body.count)
    except ValueError as e:
        raise HTTPException(status_code=400, detail=str(e))

File: frontend/src/api/fishing.js
// 청계천 낚시터 API.
import client from "./client";

export const getPond = () => client.get("/api/fishing/pond").then((r) => r.data);
export const castLine = () => client.post("/api/fishing/cast").then((r) => r.data);
export const landFish = (token) =>
  client.post("/api/fishing/land", { token }).then((r) => r.data);
export const getCollection = () =>
  client.get("/api/fishing/collection").then((r) => r.data);
export const sellFish = (fish_id, count = 1) =>
  client.post("/api/fishing/sell", { fish_id, count }).then((r) => r.data);
